src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -514,7 +526,16 @@ private void provision() throws Exception {
         }
         File plugins = new File(getHome(), "plugins");
         Files.createDirectories(plugins.toPath());
-        FileUtils.copyURLToFile(RealJenkinsRule.class.getResource("RealJenkinsRuleInit.jpi"), new File(plugins, "RealJenkinsRuleInit.jpi"));
+        if (updateRealJenkinsRuleInitPluginBaseline) {
+            try (JarFile jf = new JarFile(war);
+                    InputStream resourceAsStream = RealJenkinsRule.class.getResourceAsStream("RealJenkinsRuleInit.jpi");
```

Comment:
Actually with https://github.com/jenkinsci/jenkins-test-harness/pull/781#discussion_r1628435294 we could clean up some longstanding tech debt and get rid of the `.jpi` file in Git: just compile the one class from sources in this harness (include in `src/main/java/`) and pack the plugin on the fly.